src/lib/components/ui/TextField/__tests__/TextField.test.jsx

```diff
@@ -1,105 +1,125 @@
+import { variantPropTest } from '../../../../../../tests/propTests/variantPropTest';
+
+const mandatoryProps = {
+  label: 'label',
+};
 
 describe('rendering', () => {
-  it('renders correctly mandatory props only', () => {
-    const tree = shallow(<TextField label="label" />);
-
-    expect(shallowToJson(tree)).toMatchSnapshot();
-  });
-
-  it('renders correctly with hidden label', () => {
-    const tree = shallow(<TextField
-      isLabelVisible={false}
-      label="With hidden label"
-    />);
-
-    expect(shallowToJson(tree)).toMatchSnapshot();
-  });
-
-  it('renders correctly with small input', () => {
-    const tree = shallow(<TextField
-      inputSize={5}
-      label="Small input"
-    />);
-
-    expect(shallowToJson(tree)).toMatchSnapshot();
-  });
-
-  it('renders correctly the number input type', () => {
-    const tree = shallow(<TextField
-      label="Number input type"
-      type="number"
-    />);
-
-    expect(shallowToJson(tree)).toMatchSnapshot();
-  });
-
-  it('renders correctly the number input type with small input', () => {
-    const tree = shallow(<TextField
-      label="Number input type"
-      inputSize={1}
-      type="number"
-    />);
-
-    expect(shallowToJson(tree)).toMatchSnapshot();
-  });
-
-  it('renders correctly with custom props', () => {
-    const tree = shallow(<TextField
-      autoCapitalize="off"
-      autoComplete="off"
-      label="With custom props"
-    />);
-
-    expect(shallowToJson(tree)).toMatchSnapshot();
-  });
-
-  it('renders correctly with all props', () => {
-    const tree = shallow(<TextField
-      disabled
-      fullWidth
-      helpText="some help"
-      id="test"
-      inFormLayout
-      inputSize={20}
-      isLabelVisible={false}
-      label="All props"
-      layout="horizontal"
-      placeholder="placeholder"
-      required
-      size="large"
-      type="email"
-      validationState="invalid"
-      validationText="some error"
-      value="value"
-      variant="filled"
-    />);
-
-    expect(shallowToJson(tree)).toMatchSnapshot();
+  it.each([
+    ...disabledPropTest,
+    ...forwardedRefPropTest(React.createRef()),
+    ...fullWidthPropTest,
+    ...helpTextPropTest,
+    [
+      {
+        helpText: 'help text',
+        id: 'id',
+        validationText: 'validation text',
+      },
+      (rootElement) => {
+        expect(rootElement).toHaveAttribute('id', 'id__label');
+        expect(within(rootElement).getByTestId('id'));
+        expect(within(rootElement).getByText('label')).toHaveAttribute('id', 'id__labelText');
+        expect(within(rootElement).getByText('help text')).toHaveAttribute('id', 'id__helpText');
+        expect(within(rootElement).getByText('validation text')).toHaveAttribute('id', 'id__validationText');
+      },
+    ],
+    ...inFormLayoutPropTest,
+    [
+      { inputSize: 3 },
+      (rootElement) => expect(within(rootElement).getByRole('textbox')).toHaveAttribute('size', '3'),
```

Comment:
It may be further useful to test if `size` attribute is set for all input types except `number`.

Reply:
I added a test for `type=number`